napari/layers/labels/labels.py

```diff
@@ -1068,14 +1068,16 @@ def _raw_to_displayed(
         if labels_to_map.size == 0:
             return self._cached_mapped_labels[data_slice]
 
-        if self.color_mode == LabelColorMode.AUTO:
-            mapped_labels = _cast_labels_data_to_texture_dtype_auto(
-                labels_to_map, self._random_colormap
-            )
+        if new_label is not None and self.contour < 1:
+            if new_label not in self._cached_labels_mapping:
+                map_ = self.colormap._map_to_gpu(
+                    self._slice.image.raw.dtype.type(new_label)
+                )
+                self._cached_labels_mapping[new_label] = map_, map_.dtype
+            val, dt = self._cached_labels_mapping[new_label]
+            mapped_labels = np.full_like(labels_to_map, val, dtype=dt)
```

Comment:
The full mentioned in your question is here. 
With additional cache check.

Reply:
@Czaki I know it's in the code, I'm saying, the perfplots/targeted benchmarks you ran [above](https://github.com/napari/napari/pull/6583#issuecomment-1892769972) don't check the performance of doing this. The painting benchmarks [didn't seem to improve](https://github.com/napari/napari/pull/6583#issuecomment-1890371565) by using this strategy and as far as I can tell we have not understood that yet.